### DISCUSSION TITLE
On v7.10.0 handleSubmit after reset() gives empty data

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#5823] [Q&A]
**Describe the bug**
On 7.10.0, if I do `handleSubmit(onValid)()` right after I `reset()` the form, `handleSubmit` would give an empty object `{}` as its data

**To Reproduce**
open sandbox, hit reset button, you'll be able to see empty console logs

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-get-started-forked-h1lrz?file=/src/index.js

**Expected behavior**
`handleSubmit` should be giving the most updated form data
(Also if you change the dependency to v7.8.0 you can see the correct incoming data 🙂)

**Comment by bluebill1049** [comment]
There is an issue related to the fix at 7.8.8: https://github.com/react-hook-form/react-hook-form/pull/5634/files

I would suggest doing the following instead for reset form 

https://react-hook-form.com/api/useform/reset

```tsx
import React from "react";
import { useForm, useFieldArray, Controller } from "./src";

function App() {
  const {
    register,
    handleSubmit,
    reset,
    formState: { isSubmitSuccessful }
  } = useForm({ defaultValues: { something: "anything" } });

  const onSubmit = (data) => {
    console.log(data);
  };

  React.useEffect(() => {
    if (formState.isSubmitSuccessful) {
      reset({ something: '' });
    }
  }, [formState, submittedData, reset]);

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <input {...register("something")} />
      <input type="submit" />
    </form>
  );
}
```

<img width="998" alt="Screen Shot 2021-07-03 at 12 06 18 pm" src="https://user-images.githubusercontent.com/10513364/124340224-14f5f300-dbf7-11eb-9a0b-037e77d2acad.png">
